Panic when attempting kubicorn apply for cluster using aws profile

Following the AWS walkthrough and only got as far as `Query existing resources` with a `runtime error: invalid memory address or nil pointer dereference` related to

`kris-nova/kubicorn/cloud/amazon/public/resources/launchconfiguration.go`

It looks like an appropriate state file was generated (happy to share if helpful).

- go version go1.9.1 darwin/amd64 (also upgraded from 1.8 using brew in debugging).
AWS credentials are definitely good.
- first time installing kubicorn (today)

Output:

```bash
kubicorn apply my1stcluster
2017-10-16T15:43:25-07:00 [✔]  Selected [fs] state store
2017-10-16T15:43:25-07:00 [✔]  Loaded cluster: my1stcluster
2017-10-16T15:43:25-07:00 [✔]  Init Cluster
2017-10-16T15:43:25-07:00 [✔]  Query existing resources
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16f8df5]

goroutine 1 [running]:
github.com/kris-nova/kubicorn/cloud/amazon/public/resources.(*Lc).Actual(0xc4201e0900, 0xc420416b60, 0xb, 0xc420233b88, 0xc4202fd340, 0x0, 0x0)
	/Users/ramin/Go/src/github.com/kris-nova/kubicorn/cloud/amazon/public/resources/launchconfiguration.go:85 +0x195
github.com/kris-nova/kubicorn/cloud.(*AtomicReconciler).Actual(0xc420420880, 0xc420416b60, 0x0, 0x0, 0x0)
	/Users/ramin/Go/src/github.com/kris-nova/kubicorn/cloud/atomic_reconciler.go:49 +0x27d
github.com/kris-nova/kubicorn/cmd.RunApply(0x2438c20, 0x1ce3d38, 0x1a34060)
	/Users/ramin/Go/src/github.com/kris-nova/kubicorn/cmd/apply.go:118 +0x4c4
github.com/kris-nova/kubicorn/cmd.glob..func2(0x2430fa0, 0xc4202a95a0, 0x1, 0x1)
	/Users/ramin/Go/src/github.com/kris-nova/kubicorn/cmd/apply.go:56 +0x72
github.com/kris-nova/kubicorn/vendor/github.com/spf13/cobra.(*Command).execute(0x2430fa0, 0xc4202a9560, 0x1, 0x1, 0x2430fa0, 0xc4202a9560)
	/Users/ramin/Go/src/github.com/kris-nova/kubicorn/vendor/github.com/spf13/cobra/command.go:653 +0x2a2
github.com/kris-nova/kubicorn/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x24311e0, 0x196207c, 0x1af5f20, 0x0)
	/Users/ramin/Go/src/github.com/kris-nova/kubicorn/vendor/github.com/spf13/cobra/command.go:728 +0x2fe
github.com/kris-nova/kubicorn/vendor/github.com/spf13/cobra.(*Command).Execute(0x24311e0, 0x0, 0x563)
	/Users/ramin/Go/src/github.com/kris-nova/kubicorn/vendor/github.com/spf13/cobra/command.go:687 +0x2b
github.com/kris-nova/kubicorn/cmd.Execute()
	/Users/ramin/Go/src/github.com/kris-nova/kubicorn/cmd/root.go:85 +0x31
main.main()
	/Users/ramin/Go/src/github.com/kris-nova/kubicorn/main.go:20 +0x20
```

Thanks

--- a/cloud/amazon/public/resources/launchconfiguration.go
+++ b/cloud/amazon/public/resources/launchconfiguration.go
@@ -106,7 +106,7 @@ func (r *Lc) Expected(immutable *cluster.Cluster) (*cluster.Cluster, cloud.Resou
 		Image:            r.ServerPool.Image,
 		BootstrapScripts: r.ServerPool.BootstrapScripts,
 	}
-	if r.ServerPool.Type == cluster.ServerPoolTypeNode {
+	if r.ServerPool.Type == cluster.ServerPoolTypeNode && r.ServerPool.AwsConfiguration != nil {
 		newResource.SpotPrice = r.ServerPool.AwsConfiguration.SpotPrice
 	}
 	newCluster := r.immutableRender(newResource, immutable)

--- a/cloud/amazon/public/resources/launchconfiguration.go
+++ b/cloud/amazon/public/resources/launchconfiguration.go
@@ -81,7 +81,7 @@ func (r *Lc) Actual(immutable *cluster.Cluster) (*cluster.Cluster, cloud.Resourc
 	} else {
 		newResource.Image = r.ServerPool.Image
 		newResource.InstanceType = r.ServerPool.Size
-		if r.ServerPool.Type == cluster.ServerPoolTypeNode {
+		if r.ServerPool.Type == cluster.ServerPoolTypeNode && r.ServerPool.AwsConfiguration != nil {
 			newResource.SpotPrice = r.ServerPool.AwsConfiguration.SpotPrice
 		}
 	}

